src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let currentState = state;
+  const stateHistory = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
```

Comment:
The name `currentState` is misleading when you're actually creating a new state. Consider renaming it to `newState` or `nextState` to better reflect its purpose.

Reply:
excuse me, but i think its useless change